### PR TITLE
Refactored out proto GenerateCPP function

### PR DESCRIPTION
**Pull request by cramja**
Refactored out the GenProto methods from the main cmakelist. This cuts down on clutter and is something I ran into when making the single-node-server.

I also added some comments in the new cmake file.

@zuyu could you review?

### CODE REVIEW

File: cmake/QsProtobufGenerateCpp.cmake [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#
+# - Example
+#
+# include(QsProtobufGenerateCpp)  NOTE: already included in /CMakeLists.txt
+# QsProtobufGenerateCpp(storage_AggregationOperationState_proto_srcs
+#                       storage_AggregationOperationState_proto_hdrs
+#                       AggregationOperationState.proto)
+# Requires CMake 2.6+
+function(QS_PROTOBUF_GENERATE_CPP SRCS HDRS)
+    if(NOT ARGN)
```
Comment (zuyu) [inline cmake/QsProtobufGenerateCpp.cmake]: Please reduce to two whitespace indentations.

File: cmake/QsProtobufGenerateCpp.cmake [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        # other directories can depend on for imports.
+        string(REPLACE "/" "_" PROTOC_TARGET_NAME "protoc_${CMAKE_CURRENT_BINARY_DIR}/${FIL_WE}")
+        string(REPLACE ":" "_" PROTOC_TARGET_NAME ${PROTOC_TARGET_NAME})
+        add_custom_target(${PROTOC_TARGET_NAME} DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${FIL_WE}.pb.h")
+    endforeach()
+
+    set_source_files_properties(${${SRCS}} ${${HDRS}} PROPERTIES GENERATED TRUE)
+    set(${SRCS} ${${SRCS}} PARENT_SCOPE)
+    set(${HDRS} ${${HDRS}} PARENT_SCOPE)
+endfunction()
```
Comment (zuyu) [inline cmake/QsProtobufGenerateCpp.cmake]: Please add `EOL`.

**Comment by cramja** [comment]
@zuyu updated. I tried adding a new line but it doesn't seem to be added on github. Does it really matter? If not, I'll merge

**Comment by zuyu** [comment]
You've already added the `EOL`.

**Comment by zuyu** [comment]
LGTM!

**Comment by cramja** [comment]
What is an EOL? I thought what you were saying was to add a `\n` but is that not it?

**Comment by zuyu** [comment]
Pressing `<Enter>` adds `\n`.

**Comment by cramja** [comment]
D'oh I think what I read was `EOF` and was thinking, wtf, do I add a EOF?

**Pull request merged**